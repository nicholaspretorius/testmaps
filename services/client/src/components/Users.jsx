import React from "react";
import PropTypes from "prop-types";

const Users = props => {
  const { users, removeUser, isAuthenticated } = props;

  return (
    <div>
      <table className="table is-hoverable is-fullwidth">
        <thead>
          <tr>
            <th>Id</th>
            <th>Email</th>
            {isAuthenticated() && <th />}
          </tr>
        </thead>
        <tbody>
          {users.map(user => {
            return (
              <tr key={user.email}>
                <td>{user.id}</td>
                <td className="user-email">{user.email}</td>
                {isAuthenticated() && (
                  <td>
                    <button
                      className="button is-danger is-small"
                      onClick={() => removeUser(user.id)}
                    >
                      Delete
                    </button>
                  </td>
                )}
              </tr>
            );
          })}
        </tbody>
      </table>
    </div>
  );
};

Users.propTypes = {
  users: PropTypes.array.isRequired,
  removeUser: PropTypes.func.isRequired,
  isAuthenticated: PropTypes.func.isRequired
};

export default Users;
