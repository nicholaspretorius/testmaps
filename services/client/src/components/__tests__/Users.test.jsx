import React from "react";
import { render, cleanup } from "@testing-library/react";

import Users from "./../Users";

afterEach(cleanup);

const users = [
  {
    id: 1,
    email: "test1@test.com"
  },
  {
    id: 2,
    email: "test2@test.com"
  }
];

it("renders an email", () => {
  const { getByText } = render(
    <Users users={users} removeUser={() => true} isAuthenticated={() => true} />
  );
  expect(getByText("test1@test.com")).toHaveClass("user-email");
  expect(getByText("test2@test.com")).toHaveClass("user-email");
});

it("renders when not authenticated", () => {
  const { asFragment } = render(
    <Users users={users} removeUser={() => true} isAuthenticated={() => false} />
  );
  expect(asFragment()).toMatchSnapshot();
});

it("renders when authenticated", () => {
  const { asFragment } = render(
    <Users users={users} removeUser={() => true} isAuthenticated={() => true} />
  );
  expect(asFragment()).toMatchSnapshot();
});
