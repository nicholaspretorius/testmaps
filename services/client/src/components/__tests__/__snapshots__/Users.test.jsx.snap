// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders when authenticated 1`] = `
<DocumentFragment>
  <div>
    <table
      class="table is-hoverable is-fullwidth"
    >
      <thead>
        <tr>
          <th>
            Id
          </th>
          <th>
            Email
          </th>
          <th />
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            1
          </td>
          <td
            class="user-email"
          >
            test1@test.com
          </td>
          <td>
            <button
              class="button is-danger is-small"
            >
              Delete
            </button>
          </td>
        </tr>
        <tr>
          <td>
            2
          </td>
          <td
            class="user-email"
          >
            test2@test.com
          </td>
          <td>
            <button
              class="button is-danger is-small"
            >
              Delete
            </button>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;

exports[`renders when not authenticated 1`] = `
<DocumentFragment>
  <div>
    <table
      class="table is-hoverable is-fullwidth"
    >
      <thead>
        <tr>
          <th>
            Id
          </th>
          <th>
            Email
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            1
          </td>
          <td
            class="user-email"
          >
            test1@test.com
          </td>
        </tr>
        <tr>
          <td>
            2
          </td>
          <td
            class="user-email"
          >
            test2@test.com
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;
